sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/MeterSdkProvider.java

```diff
@@ -41,11 +45,16 @@ private MeterSdkProvider(Clock clock, Resource resource) {
 
   @Override
   public MeterSdk get(String instrumentationName) {
-    return registry.get(instrumentationName);
+    return get(instrumentationName, null);
   }
 
   @Override
-  public MeterSdk get(String instrumentationName, String instrumentationVersion) {
+  public MeterSdk get(String instrumentationName, @Nullable String instrumentationVersion) {
+    // Per the spec, both null and empty are "invalid" and a "default" should be used.
+    if (instrumentationName == null || instrumentationName.isEmpty()) {
+      LOGGER.fine("Meter requested without instrumentation name.");
+      instrumentationName = DEFAULT_METER_NAME;
```

Comment:
I think this will be a common case, I would suggest to actually resolve this as a static variable and return the instance.

Reply:
I'm not sure I understand. Are you suggesting a static `MeterSdk` to represent the default, rather than loading the cached instance from the registry? Are we OK with ignoring the potentially valid version number provided when returning the static instance?

I don't think we should use a static here. Firstly, we have the cache, and secondly, optimizing the code for what is supposed to be a "don't do this" use case seems wrong. If this will be a common case, then it feels like there should be [a stronger push on the spec](https://github.com/open-telemetry/opentelemetry-specification/issues/586) to make `null` a valid value.

Reply:
Just have a "private static Meter DEFAULT_METER = registry.get(DEFAULT_NAME, null)" and return that if no name instead of always asking the registry to return the value.

Reply:
I understand that this is a no in the specs, but I do believe users will do this a lot :)). We can do that later

Reply:
If we don't throw exception, user will use this most of the time, that is my theory

Reply:
Should we keep throwing an exception then? The instrumentation name should not be a dynamically generated string that can accidentally be null at runtime anyway.

Reply:
I drafted #2036 which goes the route of not only throwing the exception in the SDK, but also in the API. By doing so, 3rd-party library authors who consume just the API are less likely to create a library that works until the application developer drops in the SDK.

Reply:
Unfortunately, we explicitly can't do what you put into #2036. We cannot break applications that have poorly written instrumentation in them.

Reply:
OK. But can we log a SEVERE at least? The logspam should be enough pressure to fix an empty instrumentation name hopefully.

Reply:
We definitely should not spam an end-application's logs, as this can crash the app (run out of disk space, etc). And, if the end-user's application is using a library with bad instrumentation built in to it, they may not even have the option of fixing it (and they may not even be using OpenTelemetry at all!)

Reply:
My original motivation for opening this issue and submitting a PR was that I was instrumenting an internal library that another team consumes. It was unclear to me as a library author what value I was supposed to provide as the `instrumentationName`, so I tried `null`. The API accepted it without crashing! It wasn't until later, when I did some testing and added the SDK, that it crashed. That would have been the consuming team who publish the app.

As the author of a library, I _wish_ my code had failed when just the API artifact was on the classpath, or at least complained loudly about it to me. But, since the spec declares that `null` should be treated as "use a default", I made this PR to modify the SDK to be lenient (thereby matching the API).

___
I think this PR meets the requirements of the current spec. More importantly, it makes the SDK and API artifacts behave consistently on invalid inputs, so there are no surprises when the SDK is used with instrumented code. It does _not_ optimize for the case that the spec says is invalid, and it doesn't help developers adding instrumentation to "do it right". 

I'll happily try to improve documentation or samples to give both library developers and application developers a better idea of what a meaningful `instrumentationName` is for their use case, and why it is important. My "using Otel for the first time" experience is still relatively fresh in my memory.

Reply:
I'm 100% ok with the SDK not blowing up when a null instrumentation name is used. But, we should not change the API parameter annotations to make it something that is "approved", nor should we make the API throw exceptions if a null name is passed in.